tests/test_updater_with_simulator.py

```diff
@@ -79,43 +86,45 @@ def test_refresh(self):
 
         self._run_refresh()
 
-    def test_targets(self):
-        targets = {
-            "targetpath": b"content",
-            "åäö": b"more content",
-            "dir/targetpath": b"dir target content"
-        }
+    targets: DataSet = {
+        "standard case": ("targetpath", b"content", "targetpath"),
+        "non-asci case": ("åäö", b"more content", "%C3%A5%C3%A4%C3%B6"),
+        "subdirectory case": ("a/b/c/targetpath", b"dir target content", "a%2Fb%2Fc%2Ftargetpath"),
+    }
+
+    @utils.run_sub_tests_with_dataset(targets)
+    def test_targets(self, test_case_data: Tuple[str, str]):
```

Comment:
Since you've defined `class TargetData`, this type annotation is no longer true or at least it should be `Tuple[str, str, str]`. Not sure what would be the opinion of `mypy` actually.